pkg/scheduler/schedule_one.go

```diff
@@ -467,10 +467,14 @@ func (sched *Scheduler) findNodesThatFitPod(ctx context.Context, fwk framework.F
 		if !s.IsRejected() {
 			return nil, diagnosis, s.AsError()
 		}
-		// All nodes in NodeToStatusMap will have the same status so that they can be handled in the preemption.
+		// All Unschedulable nodes in NodeToStatusMap will have the same status so that they can be handled in the preemption.
 		// Some non trivial refactoring is needed to avoid this copy.
-		for _, n := range allNodes {
-			diagnosis.NodeToStatusMap[n.Node().Name] = s
+		if s.Code() == framework.Unschedulable {
+			for _, n := range allNodes {
+				diagnosis.NodeToStatusMap[n.Node().Name] = s
```

Comment:
why not absent node status here?

Reply:
With current semantics, `AbsentNodeStatus` is used for `UnschedulableAndUnresolvable` status only, as this field is used for diagnosis messages and not propagated further. Setting the `AbsentNodeStatus` for Unresolvable nodes might break post filter plugins that receive only the `NodeToStatusMap` ([here](https://github.com/kubernetes/kubernetes/blob/6a16d3ebb2dc912823cb1886df60e2ccc26b1001/pkg/scheduler/schedule_one.go#L176)).

If it would be possible to merge the map and absent status into one structure as described [above](https://github.com/kubernetes/kubernetes/pull/125954#discussion_r1671818405), it will make it possible to set the `Unschedulable` status here using the `AbsentNodeStatus` field.